### DISCUSSION TITLE
Conditional redirects / default pages in getStaticProps in case of 404

### DISCUSSION THREAD

**Discussion by tyteen4a03** [vercel/next.js#14250] [Help]
Say I have the route setup `/blog/[year]/[id]`. I load the blog posts in `getStaticProps`. However there may be times when [year] or [id] doesn't yield any results, so in case of `[id]` not existing I want to redirect to `/blog/[year]/1`, and in case of `[year]` not existing I want to redirect to `/blog/2020/1` (both of which I know definitely exists).

How do I do this in getStaticProps?

**Comment by enricoschaaf** [comment]
Do you deploy to vercel? If yes, the best thing would be to set up a custom 404 page for your specific directory. https://vercel.com/docs/configuration#routes/advanced/custom-404, where you redirect to [year]/1

**Reply by tyteen4a03** [reply]
Nope, I'm looking to deploy on my own server.

**Reply by enricoschaaf** [reply]
Have a look at this RFC #9081, I think that could help you in the future, for now you could use fallback: true, and just generate the first blogpost with id: 1, but that would just show the same content not redirect or send back a page that basically does a client side redirect to /1.  But ideally this should be handled by your server to also send back the proper status code.
